iree/test/e2e/regression/dynamic_torch_index_select_high_rank.mlir

```diff
@@ -0,0 +1,58 @@
+// RUN: iree-run-mlir %s -iree-hal-target-backends=llvm-ir -input-value="2x2xi32=[6, 7] [8, 9]" -input-value="2x2x2x2xi32=[[[0, 1] [1, 0]] [[0, 0] [1, 1]]] [[[1, 1] [0, 0]] [[0, 1] [1, 0]]]" | IreeFileCheck %s
```

Comment:
Can you point me to (or add a test) where we are testing the wrap-around (negative index) behavior?

Reply:
In tensorflow, axis and batch_dims can be negative. 
But the function `TorchIndexSelectOpConversion` in HLOToLinalgOnBuffers.cpp disable the nagtive, so modify the code and add the negative case.